rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -270,19 +270,41 @@ rmw_create_publisher(
     goto fail;
   }
 
+  switch (qos_policy.history) {
+    case RMW_QOS_POLICY_KEEP_LAST_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_KEEP_ALL_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
+      break;
+    default:
+      throw std::runtime_error("Unknown QoS history policy");
```

Comment:
I know it was a month ago, I was hoping my comment wouldn't be taken seriously, that's what the smiley was for... next time I'll use a picture of a kitty.

Anyway, I'll do the goto thing.